nemo/collections/multimodal/models/multimodal_llm/neva/neva_model.py

```diff
@@ -198,22 +198,30 @@ def replace_media_embeddings(self, input_ids, inputs_embeds, media):
         # flatten patches
         media_features = media_features.view(batch_size, -1, hidden_size)
 
-        # create an indices matrix used in torch.scatter
-        padded_media_indices = torch.ones(
-            (batch_size, num_images_per_sample), dtype=torch.long, device=input_ids.device
+        # create an indices matrix used in torch.scatter {
+        sorted_media_end_positions_mask, media_end_positions_mask_sort_idx = (
+            # NOTE: to(torch.long) is needed because PyTorch does not have sort for boolean tensors on CUDA
+            (input_ids == self.media_end_id).to(torch.long).sort(dim=-1, descending=True, stable=True)
         )
```

Comment:
I have a suspicion that stability might not necessarily be a requirement judging from the code below. Let's test the current impl and then I will try to replace `sort` with `topk(...,k=num_images_per_sample)`.